src/fairseq2/datasets/data_reader.py

```diff
@@ -104,6 +105,8 @@ def __next__(self) -> List[BatchT]:
 
             batches.append(batch)
 
+        self._total_batch_size = _total_batch_size(batches, self._gang)
```

Comment:
@cbalioglu how do you recommend I deal with this mypy issue?

Reply:
Please rebase this PR on top of latest main. I just merged #416 that introduces a `Batch` interface with a `batch_size` property.